log/logging.go

```diff
@@ -188,22 +203,112 @@ func SetTimestampFlags(flags int) {
 	DefaultLogger.SetTimestampFlags(flags)
 }
 
+type Logger interface {
+	Fatal(description string, keysAndValues ...interface{})
+	Error(description string, keysAndValues ...interface{})
+	Warn(description string, keysAndValues ...interface{})
+	Info(description string, keysAndValues ...interface{})
+	Debug(description string, keysAndValues ...interface{})
+	Trace(description string, keysAndValues ...interface{})
+
+	SetLevel(level LogLevel)
+	SetOutput(w io.Writer)
+	SetTimestampFlags(flags int)
+	SetStaticField(name string, value interface{})
+
+	logMessage(level LogLevelName, id string, description string, keysAndValues ...interface{})
```

Comment:
Not sure if anyone has pointed this out yet, but this will prevent anyone outside of this package from actually implementing this interface. Intentional?